Issue 27799 dot cli do not require the dot service yml

jgambarios opened the pull request:
Added support to use a `dotcm-url` along with the `--token` option, bypassing the required configuration setup.

jgambarios committed:
#27799 Adding support for a dotCMS URL and bypass the yml file configuration

jgambarios committed:
#27799 Adding support for a dotCMS URL and bypass the yml file configuration

jgambarios committed:
#27799 Adding support for a dotCMS URL and bypass the yml file configuration

jgambarios committed:
Merge remote-tracking branch 'origin/master' into issue-27799-dotCLI-Do-not-require-the-dot-service-yml

jgambarios committed:
#27799 Fixing IT

jgambarios committed:
Merge remote-tracking branch 'origin/master' into issue-27799-dotCLI-Do-not-require-the-dot-service-yml

jgambarios committed:
#27799 Applying sonarlint feedback

jgambarios committed:
#27799 Renaming test class

jgambarios committed:
Merge remote-tracking branch 'origin/master' into issue-27799-dotCLI-Do-not-require-the-dot-service-yml

jgambarios committed:
#27799 Handling nulls

jgambarios committed:
#27799 Improving cleanup in IT

jdotcms commented on tools/dotcms-cli/api-data-model/src/main/java/com/dotcms/api/client/model/RemoteURLParam.java:
```diff
@@ -0,0 +1,18 @@
+ * The RemoteURLParam interface is used in the {@link com.dotcms.cli.common.AuthenticationMixin} to
+ * set and get the dotCMS URL parameter. It provides methods to set and retrieve the URL.
+ */
+public interface RemoteURLParam {
```
it would be great to have interface such as Resetable

jgambarios replied:
Done, `Resetable` interface created, thanks for the feedback!

jgambarios committed:
#27799 Implemented Resetable interface in API models

Added a new Resetable interface and implemented it in the AuthenticationParam and RemoteURLParam classes. This implementation standardized the way objects reset to their original state across the API models. As a result, removed the reset method from these classes as it's now inherited from the Resetable interface.
